packages/bunnymen-core/src/node.ts

```diff
@@ -1,246 +1,294 @@
+  downloadedData: (property: string) => void
 }
 
 export class Node extends EventEmitter {
+  private _topic: string
+  private _node
+  private _libp2p
+  private _opts: any
+  private _peerId: PeerIdStr = ''
+  private _peers: string[]
+  private _currentLeader: string = ''
+  private _currentCid: string = ''
+  private _currentStep: number = 0
+  private _frequency = -1
+  private POLLING_FREQENCY: number = 100
+  private _untypedOn = this.on
+  private _untypedEmit = this.emit
 
-    private _topic: string
-    private _node
-    private _peerId: PeerIdStr = ''
-    private _peers: string[]
-    private _currentLeader: string = ''
-    private _currentCid: string = ''
-    private _currentStep:  number = 0
-    private _frequency = -1
-    private POLLING_FREQENCY: number = 100
-    private _untypedOn = this.on
-    private _untypedEmit = this.emit
-    public on = <K extends keyof INodeEvents>(event: K, listener: INodeEvents[K]): this => this._untypedOn(event, listener)
-    public emit = <K extends keyof INodeEvents>(event: K, ...args: Parameters<INodeEvents[K]>): boolean => this._untypedEmit(event, ...args)
-
-    get peerId() {
-        return this._peerId
-    }
-    get peers() {
-        return this._peers
-    }
+  public on = <K extends keyof INodeEvents>(
+    event: K,
+    listener: INodeEvents[K]
+  ): this => this._untypedOn(event, listener)
 
-    constructor(topic: string) {
-        super()
+  public emit = <K extends keyof INodeEvents>(
+    event: K,
+    ...args: Parameters<INodeEvents[K]>
+  ): boolean => this._untypedEmit(event, ...args)
 
-        const libp2pBundle = (opts: any) => {
-            this._peerId = opts.peerId.toString()
-            const bootstrapList = opts.config.bootstrap
-            return this.configureLibp2p(opts, bootstrapList)
-        }
+  get peerId() {
+    return this._peerId
+  }
+  
+  get peers() {
+    return this._peers
+  }
 
-        this._node =  IPFS.create({
-            repo:  path.join(os.tmpdir(), `repo-${nanoid()}`),
-            libp2p: libp2pBundle
-        })
+  constructor(topic: string) {
+    super()
+    this._topic = topic
+    this._peers = new Array(0)
+    this._libp2p = ((opts: any) => {
+      this._opts = opts
+      this._peerId = opts.peerId.toString()
+      const bootstrapList = opts.config.bootstrap
+      return this.configureLibp2p()
+    })
+    
+    this._node = IPFS.create({
+      repo: path.join(os.tmpdir(), `repo-${nanoid()}`),
+      libp2p: this._libp2p,
+    })
+  }
 
-        this._topic = topic
-        this._peers = new Array(0)
-    }
+  async poll(frequency: number) {
+    this._frequency = frequency
+    setInterval(async () => {
+      if (this._peerId != undefined) {
+        this.checkForNewPeers()
+        if (this._currentStep >= frequency) {
+          this.selectLeader()
+          this._currentStep = 0
+        }
+        this._currentStep += this.POLLING_FREQENCY
+      }
+    }, this.POLLING_FREQENCY)
+  }
```

Comment:
This is cool, so it polls every 100 ms, but only selects a leader at the frequency from arguments?

Reply:
yessir